app/components/ui/tale-tab-metadata/component.js

```diff
@@ -0,0 +1,91 @@
+import Ember from 'ember';
+import { later } from '@ember/runloop';
+import { computed } from '@ember/object';
+import { not } from '@ember/object/computed';
+
+import $ from 'jquery';
+
+import config from '../../../config/environment';
+
+const taleStatus = Object.create({
+  NONE: -1,
+  READ: 0,
+  WRITE: 1,
+  ADMIN: 2,
+  SITE_ADMIN: 100
+});
+
+export default Ember.Component.extend({
+  apiHost: config.apiHost,
+  environments: [],
+  model: null,
+  init() {
+    this._super(...arguments);
+    
+    // Fetch images for user to select an environment
+    let component = this;
+    $.getJSON(this.get('apiHost') + '/api/v1/image/').then(function(images) {
+      component.set('environments', images);
+      component.selectDefaultImageId();
+    });
+  },
+  
+  didRender() {
+    // Enable the environment dropdown functionality
+    $('.ui.dropdown').dropdown();
+    this.selectDefaultImageId();
+  },
+  
+  selectDefaultImageId() {
+    // Select the current imageId by default
+    let selectedImageId = this.get("model").get("tale").get("imageId")
+    $('.ui.dropdown').dropdown('set selected', selectedImageId);
+  },
+  
+  canEditTale: computed('model.tale._accessLevel', function () {
+    return this.get('model') && this.get('model').get('tale') && this.get('model').get('tale').get('_accessLevel') >= taleStatus.WRITE;
+  }).readOnly(),
+  cannotEditTale: computed.not('canEditTale').readOnly(),
+  
+  actions: {
+    // TODO: Status messages are still missing
+    updateTale() {
+      let tale = this.get("model").get("tale");
+      let component = this;
+      //component.set("tale_creating", true);
+      console.log("Sending updated tale:", tale);
```

Comment:
Small nitpicking; I believe we have an eslint `no-console` rule from the .eslintrc file that we've been trying to follow. This may have come from the controller code, but we generally try to update old code to follow the current linter settings.

Reply:
Does EmberJS offer a lint-friendly way to write to the console?

With AngularJS there was `$log` service that provided a bit more robust logging support without directly relying on `console.log`

Reply:
I know EmberJS provides the following [documentation](https://guides.emberjs.com/release/configuring-ember/debugging/) on debugging Ember apps but it looks like a fair amount of overhead.


I usually use console.log because it's quick and then remove it after I'm done.

Reply:
Interesting.. this documentation seems to be more concerned with tuning the existing logging done by Ember in the background. I guess I'm not sure why using `console.log` would be discouraged is there is no framework-specific substitute.

Although I definitely understand the desire to keep these at a minimum and to remove any leftover debug messages before merging PRs, there are certainly times (such as in the `onFail` function below) where we may want to log error messages from the server. Is there an example of the accepted pattern somewhere that I could take a look at?

Reply:
It seems rather esoteric to me. You can find a handful of blog posts that argue not to use them, and the [docs](https://eslint.org/docs/rules/no-console) don't give a reason other than `Such messages are considered to be for debugging purposes and therefore not suitable to ship to the client`.

I don't know of any better way to handle a failure from the server other than an error message popup. In more complicated parts of the codebase I could see leaving commented out logging statements in. @cyberdante, what's your opinion on `console.log`?